internal/cmdutil/utils_test.go

```diff
@@ -264,3 +265,175 @@ func TestGetSubtaskHandle(t *testing.T) {
+		expected bool
+	}{
+		{
+			name: "returns false when no_input is not set",
+			setup: func() {
+				viper.Reset()
+			},
+			expected: false,
+		},
+		{
+			name: "returns true when no_input is set to true",
+			setup: func() {
+				viper.Reset()
+				viper.Set("no_input", true)
+			},
+			expected: true,
+		},
+		{
+			name: "returns false when no_input is set to false",
+			setup: func() {
+				viper.Reset()
+				viper.Set("no_input", false)
+			},
+			expected: false,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			tc.setup()
+			assert.Equal(t, tc.expected, IsNoInputMode())
+		})
+	}
+}
+
+func TestShouldPrompt(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name          string
+		localNoInput  bool
+		globalNoInput bool
+		expected      bool
+	}{
+		{
+			name:          "should prompt when both local and global are false",
+			localNoInput:  false,
+			globalNoInput: false,
+			expected:      true,
+		},
+		{
+			name:          "should not prompt when local is true and global is false",
+			localNoInput:  true,
+			globalNoInput: false,
+			expected:      false,
+		},
+		{
+			name:          "should not prompt when local is false and global is true",
+			localNoInput:  false,
+			globalNoInput: true,
+			expected:      false,
+		},
+		{
+			name:          "should not prompt when both local and global are true",
+			localNoInput:  true,
+			globalNoInput: true,
+			expected:      false,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			viper.Reset()
+			if tc.globalNoInput {
+				viper.Set("no_input", true)
+			}
+
+			assert.Equal(t, tc.expected, ShouldPrompt(tc.localNoInput))
+		})
+	}
+}
+
+func TestNoInputConfigurationMethods(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name     string
+		setup    func()
+		cleanup  func()
+		expected bool
+	}{
+		{
+			name: "reads no_input from viper.Set (config file simulation)",
+			setup: func() {
+				viper.Reset()
+				viper.Set("no_input", true)
+			},
+			cleanup: func() {
+				viper.Reset()
+			},
+			expected: true,
+		},
+		{
+			name: "reads no_input from environment variable",
+			setup: func() {
+				viper.Reset()
+				viper.AutomaticEnv()
+				viper.SetEnvPrefix("jira")
+				os.Setenv("JIRA_NO_INPUT", "true")
+			},
+			cleanup: func() {
+				os.Unsetenv("JIRA_NO_INPUT")
+				viper.Reset()
+			},
+			expected: true,
+		},
+		{
+			name: "environment variable false is respected",
+			setup: func() {
+				viper.Reset()
+				viper.AutomaticEnv()
+				viper.SetEnvPrefix("jira")
+				os.Setenv("JIRA_NO_INPUT", "false")
+			},
+			cleanup: func() {
+				os.Unsetenv("JIRA_NO_INPUT")
+				viper.Reset()
+			},
+			expected: false,
+		},
+		{
+			name: "prefers viper.Set over unset environment variable",
+			setup: func() {
+				viper.Reset()
+				viper.AutomaticEnv()
+				viper.SetEnvPrefix("jira")
+				os.Unsetenv("JIRA_NO_INPUT")
```

Comment:
This one could be removed 

```suggestion
```

I feel like you added it for avoiding possible issues with parallel testing

Reply:
Tried to address this in da24dda.